hack/common.sh

```diff
@@ -1066,6 +1066,16 @@ function os::build::environment::withsource() {
 
   docker start "${container}" > /dev/null
   docker logs -f "${container}"
+
+  # extract content from the image
+  if [[ -n "${OS_BUILD_ENV_PRESERVE}" ]]; then
+    local workingdir="$(docker inspect -f '{{ index . "Config" "WorkingDir" }}' "${container}")"
+    IFS=: eval 'for path in ${OS_BUILD_ENV_PRESERVE}; do
```

Comment:
Setting IFS on a single line and not having side effects - there's no other
way to get IFS locked to not have to reset it.  Felt this was more side
effect free.

On Tue, Aug 16, 2016 at 12:42 PM, Steve Kuznetsov <notifications@github.com>
wrote:

> In hack/common.sh
> <https://github.com/openshift/origin/pull/10458#discussion_r74972535>:
>
> > @@ -1066,6 +1066,16 @@ function os::build::environment::withsource() {
> >
> >    docker start "${container}" > /dev/null
> >    docker logs -f "${container}"
> > +
> > +  # extract content from the image
> > +  if [[ -n "${OS_BUILD_ENV_PRESERVE}" ]]; then
> > +    local workingdir="$(docker inspect -f '{{ index . "Config" "WorkingDir" }}' "${container}")"
> > +    IFS=: eval 'for path in ${OS_BUILD_ENV_PRESERVE}; do
>
> Why is this in an eval?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10458/files/4b046250753fe0e53bf15c100edb397d81ac6315#r74972535>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p1AhEGKgfomF4qCBKrjJNa_mFpuHks5qgeh9gaJpZM4Jlmyi>
> .
>